backend/module/eCampCore/src/EntityService/CampCollaborationService.php

```diff
@@ -222,6 +267,19 @@ private function updateInvitation(CampCollaboration $campCollaboration, $data) {
             if (isset($data->role)) {
                 $campCollaboration->setRole($data->role);
             }
+            if (isset($data->inviteKey)
+                && CampCollaboration::STATUS_ESTABLISHED == $data->status
+                && CampCollaboration::STATUS_INVITED == $campCollaboration->getStatus()) {
+                if ($data->inviteKey !== $campCollaboration->getInviteKey()) {
+                    $messages['inviteKey'] = ['inviteKeyDoesNotMatch' => 'The sent inviteKey does not match'];
+
+                    throw (new EntityValidationException())->setMessages($messages);
+                }
+                $campCollaboration->setInviteKey(null);
+                $campCollaboration->setInviteEmail(null);
+                $campCollaboration->setUser($authUser);
+                $campCollaboration->setStatus(CampCollaboration::STATUS_ESTABLISHED);
+            }
```

Comment:
Maybe we should do this in an RPC endpoint rather than a REST endpoint, or what do you think?

Reply:
This might be a good idea.
But when do we stop using REST and Start using RPC? After the first problem?

Reply:
Moved to RPC